csharp/test/Apache.Arrow.IntegrationTest/JsonFile.cs

```diff
@@ -25,12 +33,199 @@ public class JsonFile
         public JsonSchema Schema { get; set; }
         public List<JsonRecordBatch> Batches { get; set; }
         //public List<DictionaryBatch> Dictionaries {get;set;}
+
+        public static async ValueTask<JsonFile> ParseAsync(FileInfo fileInfo)
+        {
+            using var fileStream = fileInfo.OpenRead();
+            var options = GetJsonOptions();
+            return await JsonSerializer.DeserializeAsync<JsonFile>(fileStream, options);
+        }
+
+        public static JsonFile Parse(FileInfo fileInfo)
+        {
+            using var fileStream = fileInfo.OpenRead();
+            var options = GetJsonOptions();
+            return JsonSerializer.Deserialize<JsonFile>(fileStream, options);
+        }
+
+        private static JsonSerializerOptions GetJsonOptions()
+        {
+            JsonSerializerOptions options = new JsonSerializerOptions()
+            {
+                PropertyNamingPolicy = JsonFileNamingPolicy.Instance,
+            };
+            options.Converters.Add(new ValidityConverter());
+            return options;
+        }
     }
 
     public class JsonSchema
     {
         public List<JsonField> Fields { get; set; }
         public JsonMetadata Metadata { get; set; }
+
+        /// <summary>
```

Comment:
Note that all this below is moved from `IntegrationCommand.cs` in order for it to be available to the integration testing harness.